Services/Catalogue/Mappers/ProductsMapper.php

```diff
@@ -223,6 +224,133 @@ public function map($product)
         return $mappedProduct;
     }
 
+    public function addAttribute(
+        &$product,
+        $id,
+        $name,
+        array $values,
+        $type = "string",
+        $isSearchable = false,
+        $isFilterable = false,
+        $isVariantAware = false,
+        $addInAutoComplete = false,
+        $autocompletePosition = "front",
+        $addInAutocompleteIndividually = false,
+        $autocompleteGlue = ""
+    ) {
+
+        $attribute = [
+            'id'   => $id,
+            'type' => $type,
+            'name' => $name,
+            'addInAutocomplete' => $addInAutoComplete,
+            'autocompletePosition' => $autocompletePosition,
+            'autocompleteGlue' => $autocompleteGlue,
+            'addInAutocompleteIndividually' => $addInAutocompleteIndividually,
+            'isSearchable'=> $isSearchable,
+            'isFilterable'=> $isFilterable,
+            'values' => [],
+        ];
+
+        if (isset($product['attributes'][$attribute['id']])) {
+            $attribute = $product['attributes'][$attribute['id']];
+        }
+
+        foreach ($values as $aValue) {
+            $valueAttribute = [
+                'value' => [
+                    $aValue
+                ],
+                'variationId' => $product['id'],
+                'inStock' => $product['inStock'],
+            ];
+
+            if (is_array($aValue) && $isVariantAware && isset($aValue['value']) && count($aValue['value'])) {
+                foreach ($aValue as $keyValueAttribute => $valueValueAttribute) {
+                    $valueAttribute[$keyValueAttribute] = $valueValueAttribute;
+                }
+                $attribute['values'][] = $valueAttribute;
+            } else {
+                $attribute['values'][$aValue] = $valueAttribute;
+            }
+        }
+
+        $attribute['values'] = array_values($attribute['values']);
+
+        $product['attributes'][$attribute['id']] = $attribute;
+    }
+
+    private function addKSAAttribute(&$mappedProduct)
```

Comment:
```suggestion
    private function getKeySizesCount(&$mappedProduct)
```

Also keep this function in `Services/Catalogue/Mappers/ConfigurableProductsData.php`